clickhouse_driver/client.py

```diff
@@ -176,7 +191,8 @@ def execute_iter(
             )
 
         except Exception:
-            self.connection.disconnect()
+            self.disconnect()
+            self.reset_last_query()
```

Comment:
Last query resetting can be gently placed into client's `disconnect` method.

Reply:
Moved, calling `disconnect` will now reset the last query.